### DISCUSSION TITLE
How to use SVG as a component?

### DISCUSSION THREAD

**Discussion by cnguu** [quasarframework/quasar#15864] [General - Components / Directives / etc] [answered]
example input:

```vue
<script setup lang="ts">
  import ASvg from 'assets/svg/404.svg';

  defineOptions({ name: 'ResultException' });
</script>

<template>
  <div class="flex-col-center gap-24px min-h-520px wh-full overflow-hidden">
    <div class="flex text-400px text-primary">
      <!-- use here -->
      <ASvg />
    </div>
  </div>
</template>
```

but I get some error:

![image](https://github.com/quasarframework/quasar/assets/10758182/cf65beb3-035f-4965-b3ed-048cc9ee707f)

support output:

```
<div>
  <svg>
    <!-- other dom -->
  </svg>
</div>
```

devDependencies:
```
"@quasar/app-webpack": "3.9.2",
"@quasar/cli": "2.2.0",
"webpack": "5.82.1"
```

**Comment by suleimanas** [comment, marked as answer]
- you can use Qimg or the img tag and load your SVG file with the src attribute
- or you can make your own component and put the full `<svg>...</svg> `
- or use vue-svg-loader

but you're just loading a 404 img I recommend the first method

**Reply by cnguu** [reply]
Thanks for your answer.
- I need to change the color dynamically, so I can't use the img tag to load the svg.
- I tried using vue-svg-loader, but it didn't work in quasar.
- Finally, I used template wrapped svg as a vue component, it doesn't look elegant enough, but it works.
